virtcontainers/kata_agent.go

```diff
@@ -136,8 +136,18 @@ func (k *kataAgent) generateVMSocket(id string, c KataAgentConfig) error {
 	if c.UseVSock {
 		// We want to go through VSOCK. The VM VSOCK endpoint will be our gRPC.
 		k.Logger().Debug("agent: Using vsock VM socket endpoint")
-		// We dont know yet the context ID - set empty vsock configuration
-		k.vmSocket = kataVSOCK{}
+
```

Comment:
Please check vmSocket is not set here so that we never allocate vsock cid twice for one sandbox.

Reply:
Also, could you add a unit test to assert the expected new behaviour?

Reply:
Both comments addressed, thanks.